### PR TITLE
From passing zero tests to 1,153 out of an unknown number of tests for nn.functional.max_pool2d

### PR DESCRIPTION
**Pull request by hosseinsarshar**
This PR is a partial fix for the nn.functional.max_pool2d function - this function is aggressively tested by passing a variation of parameters - it started with passing zero tests, now it passes 1153 tests - creating this PR to keep the changes up to here - 

Will continue fixing for the rest of tests for this function and other variations (1d and 3d):

```
input=XLATensor2(<class 'jaxlib.xla_extension.ArrayImpl'> [[[-1.7725346  -1.236596   -7.7467732  -4.3070955   3.615653
   -6.730438  ]
  [ 6.0401974   4.662748   -5.374122   -3.8238592  -7.2418194
    5.582711  ]
  [ 2.0260687  -4.4604063   3.068091    0.96788585  8.458886
   -0.3962202 ]]

 [[-8.515981    5.5112133   5.3879347   3.7552953  -8.1066885
    1.7102215 ]
  [ 3.3532753  -1.5467752   3.7327058  -7.3867993  -8.166386
   -7.483256  ]
  [-8.213648   -3.7317863   3.0662498   1.7091808   8.218179
    5.483536  ]]])
args=()
kwargs={'kernel_size': 3, 'stride': 2, 'ceil_mode': True, 'padding': 0, 'dilation': 1, 'return_indices': True}
window_shape_full=(1, 1, 3, 3)
```

Expected output:
```
(tensor([[[6.0402, 8.4589, 8.4589]],

        [[5.5112, 8.2182, 8.2182]]]), tensor([[[ 6, 16, 16]],

        [[ 1, 16, 16]]]))
```

Given output:
```
(tensor([[[[6.0402, 8.4589, 8.4589]],

         [[5.5112, 8.2182, 8.2182]]]]), tensor([[[[ 6, 16, 16]],

         [[ 1, 16, 16]]]]))
```

Error:
`AssertionError: The values for attribute 'shape' do not match: torch.Size([1, 2, 1, 3]) != torch.Size([2, 1, 3]).`

### CODE REVIEW

File: experimental/torch_xla2/torch_xla2/ops/jaten.py [hunk 1]
```diff
@@ -1159,126 +1159,117 @@ def _aten_relu(self):
...
+  else:
+      padding = list(padding)
+  
+  # If dilation is an int, make it a tuple
+  if isinstance(dilation, int):
+      dilation = (dilation,) * len(kernel_size)
+  else:
+      dilation = tuple(dilation)
+  
+  if inputs.ndim == 4:
```
Comment (qihqi) [inline experimental/torch_xla2/torch_xla2/ops/jaten.py:1214]: Looks like you want to reshape to rank = 4?
`inputs = jnp.expand_dims(input, jnp.arange( 4 - inputs.ndim))`

**Review by qihqi — APPROVED** [review]

**Comment by qihqi** [comment]
_ceil_padding_mode is seems to be used in other places, make sure those passes. Thanks

File: experimental/torch_xla2/torch_xla2/ops/jaten.py (hunk 1 above)
Comment (hosseinsarshar) [reply experimental/torch_xla2/torch_xla2/ops/jaten.py:1214]: that's correct - well pointed, going to incorporate it

**Comment by hosseinsarshar** [comment]
@qihqi in addition to the changes to the `_ceil_mode_padding` function, the changes causes the `max_pool2d_with_indices`, and `max_pool3d_with_indices` which are using the same implementation. Thanks for the comments.

**Pull request closed without merging**